app/join_record.ts

```diff
@@ -0,0 +1,127 @@
+"use server";
+// import { access, constants, appendFileSync, readFile } from "node:fs";
+import {
+  S3Client,
+  PutObjectCommand,
+  GetObjectCommand,
+} from "@aws-sdk/client-s3";
+import { JoinFormValues } from "@/components/JoinForm/JoinForm";
+import { Readable } from "stream";
+import { JoinRecord } from "./types";
+
+// const JOIN_FORM_LOG = process.env.JOIN_FORM_LOG as string;
+
+class JoinRecordS3 {
+  private s3Client: S3Client;
+
+  private S3_REGION: string;
+  private S3_ENDPOINT: string;
+  private S3_BUCKET_NAME: string;
+  private S3_BASE_NAME: string;
+  private S3_ACCESS_KEY: string;
+  private S3_SECRET_KEY: string;
+
+  constructor() {
+    this.S3_REGION = process.env.S3_REGION as string;
+    this.S3_ENDPOINT = process.env.S3_ENDPOINT as string;
+    this.S3_BUCKET_NAME = process.env.S3_BUCKET_NAME as string;
+    this.S3_BASE_NAME = process.env.S3_BASE_NAME as string;
+    this.S3_ACCESS_KEY = process.env.S3_ACCESS_KEY as string;
+    this.S3_SECRET_KEY = process.env.S3_SECRET_KEY as string;
+
+    // Setup the S3 client
+    this.s3Client = new S3Client({
+      region: this.S3_REGION != undefined ? this.S3_REGION : "us-east-1",
+      endpoint:
+        this.S3_ENDPOINT != undefined
+          ? this.S3_ENDPOINT
+          : "https://s3.us-east-1.amazonaws.com",
+      credentials: {
+        accessKeyId: this.S3_ACCESS_KEY,
+        secretAccessKey: this.S3_SECRET_KEY,
+      },
```

Comment:
nit: I believe these will be read automatically by the `S3Client` if you change
`S3_ACCESS_KEY` -> `AWS_ACCESS_KEY_ID`
`S3_SECRET_KEY` -> `AWS_SECRET_ACCESS_KEY`
`S3_REGION` -> `AWS_REGION`
and don't set them explicitly here

The reason this could be worthwhile (other than reducing line count) is that it makes the code more portable, if for example this was run in a Lambda function or EC2 instance, using the env-var auto-detection would allow this code to work in the new env with no changes